cmd/server/api/endpoints.go

```diff
@@ -67,8 +78,72 @@ func (a *API) newSimulation(w http.ResponseWriter, r *http.Request) {
 // This will start the simulation with the specified ID for
 // a certain number of steps.
 func (a *API) start(w http.ResponseWriter, r *http.Request) {
-	fmt.Println("Start Simulation Request")
-	w.WriteHeader(http.StatusNotImplemented)
+	log.Println(r.URL.Path)
+
+	a.mutex.RLock()
```

Comment:
Although this will acquire the read lock for the whole function. It might be better to only lock and unlock when needed.
I only think this because there is, even though in a goroutine, a write lock and this will be blocked by a read lock.

Reply:
I see your point. Even if it won't be too much time consuming, it will be more efficient the way said.
I'm going to change that.